src/sysext/sysext.c

```diff
@@ -758,6 +703,12 @@ static int verb_merge(int argc, char **argv, void *userdata) {
         if (r < 0)
                 return log_error_errno(r, "Failed to discover extension images: %m");
 
+        HASHMAP_FOREACH(img, images) {
+                r = image_read_metadata(img);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to read metadata for image %s: %m", img->name);
+        }
```

Comment:
why do this here?

since this loop is used multiple times sounds like something for a helper function?